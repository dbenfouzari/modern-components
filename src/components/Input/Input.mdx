---
name: Input
---

import { Playground, PropsTable } from 'docz';
import Input from './';

# Input

<PropsTable of={Input} />

## Basic usage

<Playground>
    {() => {
        class Example extends React.Component {
            constructor (props) {
                super(props);
                this.state = {
                    value: "Hello World !"
                }
                this.handleChange = this.handleChange.bind(this);
            }

            handleChange(nextValue) {
                this.setState({ value: nextValue });
            }
            
            render () {
                return (
                    <Input value={this.state.value} onChange={this.handleChange} />
                )
            }
        }
        
        return <Example />;
    }}
</Playground>
