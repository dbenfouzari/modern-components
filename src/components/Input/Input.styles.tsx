import darken from "polished/lib/color/darken";
import styled, { css } from "../../styled-components";

export const StyledInput = styled.input`
  border-radius: 4px;
  border: 0 solid;
  border-bottom-width: 1px;
  font-size: 14px;
  padding: 5px 10px;
  outline: none;
  background-color: #fff;
  color: ${(props) => props.theme.primaryColor};

  &:focus,
  &:active {
    border-bottom-width: 3px;
  }
`;
