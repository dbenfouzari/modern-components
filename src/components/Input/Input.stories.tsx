import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { StateDecorator, Store } from "@sambego/storybook-state";

import Input from '.';

const store = new Store({
  value: 'Hello world !'
});

storiesOf('Input', module)
  .addDecorator(StateDecorator(store))
  .add('with text', () => {
    const handleChange = (nextValue: any) => {
      action('inputChange')(nextValue);
      store.set({ value: nextValue });
    }

    return (
      <Input value='Hello world !' onChange={handleChange} />
    );
  });
