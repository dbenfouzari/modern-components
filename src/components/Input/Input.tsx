import * as React from "react";
import { StyledInput } from "./Input.styles";

interface InputProps<T> {
  /**
   * @since 0.0.1
   * @version 0.0.1
   *
   * Value passed to the input
   */
  value: T;
  /**
   * @since 0.0.1
   * @version 0.0.1
   *
   * The action that will be called when user types some text
   */
  onChange: (nextValue: T) => void;
  /**
   * @since 0.0.1
   * @version 0.0.1
   *
   * Should show as raised ?
   *
   * @default "text"
   */
  type?: "text" | "number" | "password";
}

/**
 * @since 0.0.1
 * @version 0.0.1
 * @author Donovan BENFOUZARI <d.benfouzari@gmail.com>
 *
 * This is a simple input
 */
const Input = ({
  value,
  onChange,
  type = "text"
}: InputProps<string | number>) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChange(event.currentTarget.value);
  };

  return <StyledInput type={type} value={value} onChange={handleChange} />;
};

export default Input;
