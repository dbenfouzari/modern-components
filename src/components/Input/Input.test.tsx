import * as React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "../../styled-components";
import defaultTheme from "../../theme";
import Input from "./Input";

describe("Input", () => {
  it("renders correctly", () => {
    const wrapper = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Input value="test" onChange={() => null}></Input>
        </ThemeProvider>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
