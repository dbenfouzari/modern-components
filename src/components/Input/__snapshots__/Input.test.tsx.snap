// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
.c0 {
  border-radius: 4px;
  border: 0 solid;
  border-bottom-width: 1px;
  font-size: 14px;
  padding: 5px 10px;
  outline: none;
  background-color: #fff;
  color: #28bcd6;
}

.c0:focus,
.c0:active {
  border-bottom-width: 3px;
}

<input
  className="c0"
  onChange={[Function]}
  type="text"
  value="test"
/>
`;
